### PR TITLE
tree-wide: use UINT64_MAX or friends

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/cgtop/cgtop.c [hunk 1]
```diff
@@ -978,7 +978,7 @@ static int run(int argc, char *argv[]) {
                 if (arg_batch)
                         (void) usleep(last_refresh + arg_delay - t);
```
Comment (poettering) [inline src/cgtop/cgtop.c]: what about this one?

File: src/cgtop/cgtop.c [hunk 2]
```diff
@@ -1056,7 +1056,7 @@ static int run(int argc, char *argv[]) {
                         if (arg_delay < USEC_PER_SEC)
                                 arg_delay += USEC_PER_MSEC*250;
```
Comment (poettering) [inline src/cgtop/cgtop.c]: and this one?

File: src/cryptsetup/cryptsetup.c [hunk 3]
```diff
@@ -1503,10 +1503,7 @@ static int run(int argc, char *argv[]) {
 
                 flags = determine_flags();
 
-                if (arg_timeout == USEC_INFINITY)
-                        until = 0;
-                else
-                        until = now(CLOCK_MONOTONIC) + arg_timeout;
+                until = usec_add(now(CLOCK_MONOTONIC), arg_timeout);
```
Comment (poettering) [inline src/cryptsetup/cryptsetup.c:1506]: The ask_password_xyz() functions currently take an until param of 0 as meaning "wait forever". hence this changes behaviour quite a bit. That said, those functions should probably be fixed to make "0" mean, don#t wait at all, and USEC_INFINITY mean wait forever

File: src/shared/format-table.h [hunk 4]
```diff
@@ -101,8 +101,10 @@ void table_set_width(Table *t, size_t width);
 void table_set_cell_height_max(Table *t, size_t height);
 int table_set_empty_string(Table *t, const char *empty);
 int table_set_display_all(Table *t);
-int table_set_display(Table *t, size_t first_column, ...);
-int table_set_sort(Table *t, size_t first_column, ...);
+int table_set_display_internal(Table *t, size_t first_column, ...);
+#define table_set_display(...) table_set_display_internal(__VA_ARGS__, (size_t) SIZE_MAX)
+int table_set_sort_internal(Table *t, size_t first_column, ...);
+#define table_set_sort(...) table_set_sort_internal(__VA_ARGS__, (size_t) SIZE_MAX)
```
Comment (poettering) [inline src/shared/format-table.h:107]: ah, yes, i love it. This was too easy to fuck up

**Review by poettering — CHANGES_REQUESTED** [review]
looks good, some comments

**Comment by yuwata** [comment]
@poettering Thank you for the review. The all above points are addressed. PTAL.

**Comment by yuwata** [comment]
Rebased.

File: src/shared/format-table.h [hunk 5]
```diff
@@ -101,8 +101,10 @@ void table_set_width(Table *t, size_t width);
 void table_set_cell_height_max(Table *t, size_t height);
 int table_set_empty_string(Table *t, const char *empty);
 int table_set_display_all(Table *t);
-int table_set_display(Table *t, size_t first_column, ...);
-int table_set_sort(Table *t, size_t first_column, ...);
+int table_set_display_internal(Table *t, size_t first_column, ...);
+#define table_set_display(...) table_set_display_internal(__VA_ARGS__, (size_t) SIZE_MAX)
```
Comment (keszybz) [inline src/shared/format-table.h:105]: I think the cast here is unnecessary. After all, SIZE_MAX better be of size_t.

File: src/basic/audit-util.h [hunk 6]
```diff
@@ -5,7 +5,7 @@
 #include <stdint.h>
 #include <sys/types.h>
 
-#define AUDIT_SESSION_INVALID ((uint32_t) -1)
+#define AUDIT_SESSION_INVALID (UINT32_MAX)
```
Comment (keszybz) [inline src/basic/audit-util.h:8]: UINT32_MAX already has it's own set of ().

**Review by keszybz — COMMENTED** [review]
LGTM. I'll submit a follow-up patch to remove some parens from around SIZE_MAX and friends.

**Pull request merged**